include/alp/algorithms/householder_tridiag.hpp

```diff
@@ -73,82 +103,125 @@ namespace alp {
 				// v = v - alpha * e1
 				// v = v / norm ( v )
-				Vector< D, structures::General, Dense > v;
-				rc = set( v, get_view( RR, utils::range( k + 1, n ), k ) );
 
-				Scalar< D > alpha;
+				auto v_view = get_view( RR, k, utils::range( k + 1, n ) );
+				Vector< D, structures::General, Dense > v( n - ( k + 1 ) );
+				rc = set( v, v_view );
+				if( rc != SUCCESS ) {
+					std::cerr << " set( v, view ) failed\n";
+					return rc;
+				}
+
+				Scalar< D > alpha( zero );
 				rc = norm2( alpha, v, ring );
+				if( rc != SUCCESS ) {
+					std::cerr << " norm2( alpha, v, ring ) failed\n";
+					return rc;
+				}
 
 				rc = eWiseLambda(
-							[ &v, &alpha ]( const size_t i ) {
-								if ( i == 0 ) {
-									Scalar< D > norm_v0( std::abs( v[ i ] ) );
-									foldl(alpha, v [ i ], ring.getMultiplicativeOperator() );
-									foldl(alpha, norm_v0, divide );
-									foldl(v [ i ], alpha, minus );
-								}
-							},
-							v );
-
-				Scalar< D > norm_v;
+					[ &alpha, &ring, &divide, &minus ]( const size_t i, D &val ) {
+						if ( i == 0 ) {
+							Scalar< D > norm_v0( std::abs( val ) );
+							internal::foldl( *alpha, val, ring.getMultiplicativeOperator() );
+							internal::foldl( *alpha, *norm_v0, divide );
+							internal::foldl( val, *alpha, minus );
+						}
+					},
+					v
+				);
+				if( rc != SUCCESS ) {
+					std::cerr << " eWiseLambda( lambda, v ) failed\n";
+					return rc;
+				}
+
+				Scalar< D > norm_v( zero );
 				rc = norm2( norm_v, v, ring );
+				if( rc != SUCCESS ) {
+					std::cerr << " norm2( norm_v, v, ring ) failed\n";
+					return rc;
+				}
 
 				rc = foldl(v, norm_v, divide );
+#ifdef DEBUG
+				print_vector( " v = ", v );
+#endif
 				// ===== End Computing v =====
 
-
 				// ===== Calculate reflector Qk =====
 				// Q_k = identity( n )
-				Matrix< D, structures::Symmetric, Dense > Qk( m );
-				rc = set( Qk, structures::constant::I( m ) );
+				Matrix< D, SymmOrHermType, Dense > Qk( n );
+				rc = alp::set( Qk, zero );
+				auto Qk_diag = alp::get_view< alp::view::diagonal >( Qk );
+				rc = rc ? rc : alp::set( Qk_diag, one );
 
-				Matrix< D, structures::Symmetric, Dense > vvt( m );
-				rc = set(vvt, zero );
-				// vvt = v * v^T
-				rc = outer( vvt, v, v, ring );
+				// this part can be rewriten without temp matrix using functors
+				Matrix< D, SymmOrHermType, Dense > vvt( m );
 
+				rc = rc ? rc : set( vvt, outer( v, ring.getMultiplicativeOperator() ) );
 				// vvt = 2 * vvt
-				rc = foldr( Scalar< D >( 2 ), vvt, ring.getMultiplicativeOperator() );
+				rc = rc ? rc : foldr( Scalar< D >( 2 ), vvt, ring.getMultiplicativeOperator() );
+
+
+#ifdef DEBUG
+				print_matrix( " vvt ", vvt );
+#endif
 
 				// Qk = Qk - vvt ( expanded: I - 2 * vvt )
-				rc = foldl( Qk, vvt, minus );
+				auto Qk_view = get_view( Qk, utils::range( k + 1, n ), utils::range( k + 1, n ) );
+				//conjugate view should replace this check
+				if ( grb::utils::is_complex< D >::value ) {
```

Comment:
I think since a couple of PRs ago we do have a `conjugate` view :)

Reply:
An old comment, I don't need to conjugate here as I took vector v from row rather than from column (it's already conjugated).